### PR TITLE
Create plugin for Scala publishing conventions

### PR DESCRIPTION
**Pull request by franciscodr**
This pull request creates a custom plugin to gather all the settings to publish Scala code to Maven Central. It includes:

- Generate Javadoc jar from the output of the `scaladoc` task
- Generate POM file
- Sign the artifacts

### CODE REVIEW

**Review by juanpedromoreno — APPROVED** [review]
Awesome, thanks @franciscodr !

File: buildSrc/src/main/kotlin/ScalaPublishingConventionsPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.kotlin.dsl.get
+import org.gradle.kotlin.dsl.register
+import org.gradle.plugins.signing.SigningExtension
+
+class ScalaPublishingConventionsPlugin : Plugin<Project> {
+  override fun apply(project: Project): Unit = project.run {
+    val scaladocJarTask: TaskProvider<Jar> = tasks.register<Jar>("scaladocJar") {
+      group = BasePlugin.BUILD_GROUP
+      tasks.findByName("scaladoc")?.let { dependsOn(it) }
```
Comment (nomisRev) [inline buildSrc/src/main/kotlin/ScalaPublishingConventionsPlugin.kt:18]: What if `scaladoc` is not available? 🤔 Does this task still do anything?

Alternatively, we can only add `scaladocJar` if `scaladoc` is available or I would use `requireNotNull` with a meaningful message here. (I am not entirely sure if we should throw `IllegalStateExeption` in Gradle though, I think we need to use the proper reporting tools).

**Review by nomisRev — DISMISSED** [review]
Looks great @franciscodr 🙌 Just one question/doubt.

File: buildSrc/src/main/kotlin/ScalaPublishingConventionsPlugin.kt (hunk 1 above)
Comment (franciscodr) [reply buildSrc/src/main/kotlin/ScalaPublishingConventionsPlugin.kt:18]: Good point. I guess we should control all the required settings. For instance, the signing key is mandatory when publishing to Maven Central. Let me think about it 🤔

**Review by raulraja — APPROVED** [review]

**Review by nomisRev — APPROVED** [review]

**Review by Yawolf — APPROVED** [review]

**Pull request merged**